packages/vite/src/node/plugins/asset.ts

```diff
@@ -27,17 +27,21 @@ const assetHashToFilenameMap = new WeakMap<
   ResolvedConfig,
   Map<string, string>
 >()
+// save hashes of the files that has been emitted in build watch
+const emittedHashesSet: Set<string> = new Set()
```

Comment:
Good catch, thanks for this PR. It works great.
I think that we should change `emmittedHashesSet` to be `WeakMap<ResolvedConfig, Set<string>>` though. So it follows the pattern used in other caches, as it allows users to use the same plugin for different configurations (I think there are a few other plugins that should be modified though for this to fully work, but better if we keep building towards this possibility)

Reply:
You are right, I will modify this and add a test today